docs/architecture.md

```diff
@@ -112,71 +112,199 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+That means that for each query, we take its definition, convert it into a mango predicate, and run it against the store's document collection.
+For each query result, we check if there is any change in the known ids list, and update it accordingly.
 
-`queries` has an autoUpdater mechanism that does something we can explain this way:
-- it takes the mutated documents (newly created or updated) 
-- it converts the initial `query` to a "`js predicate`" (thanks to the [sift library](https://github.com/crcn/sift.js/))
-- For each query already in the `slice` it runs this `js predicate` and detects if the query
-is concerned by the mutation
-- If the query is concerned, then it checks if it has to remove / add the id of the `mutated` 
-document
-(for now the work is done mainly in queries.js/updateData())
+Thanks to this, we ensure that all the queries are always up-to-date, and apps can be effectively rendered thanks to the [react integration](/docs/react-integration.md).
 
-So in our previous example our `todoslist` is concerned by the addition of the new todo, then 
-the `id` is added to `todolist` data, then the component linked will be refreshed with this new 
-document.
 
-### About updates
+#### Fetch policy and query naming
 
-As already mentioned, when a mutated document already exists in the store throught its `_id`, it is updated with the new content. However, this update never removes existing fields, it only merges or adds new fields. This is typically useful when two queries retrieve the same document, but not with all the fields. 
+The fetch policy is a simple mechanism useful to prevent excessive fetching. 
+It is optionnaly defined for each query and define how many time should be elapsed since the last query execution before running again the query.
+
+In this example we query the checked todos, only if the last execution occured more than 10 minutes ago:
+```js
+const queryDef = Q('io.cozy.todos').where(checked: true)
+const queryOptions = {
+  as: 'checkedTodos', // query unique name, to find it in the store
+  fetchPolicy: CozyClient.fetchPolicies.olderThan(10 * 60 * 1000) // fetch policy, for 10min 
+}
+await client.query(queryDef, queryOptions)
+```
+
+If the query had been run before the 10 minutes, an early return will happen and nothing will be updated in the store.
+
+💡 Thanks to the auto-query updater, the data will not be stale if there is a data change in the app: indeed, any change on the documents will trigger the in-memory re-evaluation of the query and a render on the app if something is different.
+
+ℹ️ The `as` option is used to name the query, which is always a good idea to ensure that it has a unique name. When it's not specified, a random id is generated, resulting in potential query duplication in the store. Likewise, one must be very careful to prevent queries with the same name, as it would mixup results in the store.
```

Comment:
I would add a like to the devtool page here: `You can easily view each query using [Devtools](https://github.com/cozy/cozy-client/blob/master/docs/devtools.md) to identify duplicates.`

Reply:
Ah, good point, I always forget its existence

Reply:
We should also add a link to the cozy-guidelines here.